malcolm/modules/pandablocks/controllers/pandablockcontroller.py

```diff
@@ -56,68 +71,74 @@ def make_meta(subtyp, description, tags, writeable=True, labels=None):
 
 
 class PandABlockController(builtin.controllers.BasicController):
-    def __init__(self,
-                 client,  # type: AClient
-                 mri_prefix,  # type: AMri
-                 block_name,  # type: ABlockName
-                 block_data,  # type: ABlockData
-                 doc_url_base,  # type: ADocUrlBase
-                 ):
-        # type: (...) -> None
-        super(PandABlockController, self).__init__(
-            mri="%s:%s" % (mri_prefix, block_name))
+    def __init__(
+        self,
+        client: AClient,
+        mri_prefix: AMri,
+        block_name: ABlockName,
+        block_data: ABlockData,
+        doc_url_base: ADocUrlBase,
+    ) -> None:
+        super().__init__(
+            mri="%s:%s" % (mri_prefix, block_name)
+        )
         # Store
         self.client = client
         self.mri_prefix = mri_prefix
         self.block_name = block_name
         self.block_data = block_data
         self.doc_url_base = doc_url_base
         # {field_name: part}
-        self.field_parts = {}  # type: Dict[str, Optional[ChangeHandler]]
+        self.field_parts: Dict[str, Optional[ChangeHandler]] = {}
         # {field_name: attr.meta}
-        self.mux_metas = {}  # type: Dict[str, VMeta]
+        self.mux_metas: Dict[str, VMeta] = {}
         # Make an icon, label and help for the Block
-        self.icon_part = self._make_common_parts()  # type: PandAIconPart
+        self.icon_part: PandAIconPart = self._make_common_parts()
         # Create parts for each field
         for field_name, field_data in block_data.fields.items():
             self._make_parts_for(field_name, field_data)
 
-    def handle_changes(self, changes, ts):
-        # type: (Dict[str, Any], TimeStamp) -> None
+    def handle_changes(
+        self, changes: Union[Dict[str, Any], Sequence[Tuple[str, str]]], ts: TimeStamp
```

Comment:
This was because the variable `changes` was being re-used in PandAManagerController.handle_changes(), mypy assumed the original definition applies to each instance of the variable in the scope.

So I have fixed it by renaming the variable used in the manager controllers method to:
```
for block_name, block_changes_values in block_changes.items():
self._child_controllers[block_name].handle_changes(block_changes_values, ts)
```